pinecone/data/index.py

```diff
@@ -81,25 +82,30 @@ def upsert_numpy_deprecation_notice(context):
     warn_deprecated(message, deprecated_in="2.2.1", removal_in="3.0.0")
 
 
-class Index(ApiClient):
+class Index():
 
     """
     A client for interacting with a Pinecone index via REST API.
     For improved performance, use the Pinecone GRPC index client.
     """
 
-    def __init__(self, index_name: str, pool_threads=1):
-        openapi_client_config = copy.deepcopy(Config.OPENAPI_CONFIG)
-        openapi_client_config.api_key = openapi_client_config.api_key or {}
-        openapi_client_config.api_key["ApiKeyAuth"] = openapi_client_config.api_key.get("ApiKeyAuth", Config.API_KEY)
-        openapi_client_config.server_variables = openapi_client_config.server_variables or {}
-        openapi_client_config.server_variables = {
-            **{"environment": Config.ENVIRONMENT, "index_name": index_name, "project_name": Config.PROJECT_NAME},
-            **openapi_client_config.server_variables,
-        }
-        super().__init__(configuration=openapi_client_config, pool_threads=pool_threads)
-        self.user_agent = get_user_agent()
-        self._vector_api = VectorOperationsApi(self)
+    def __init__(self, api_key: str, host: str, pool_threads=1, **kwargs):
+        api_key = api_key or kwargs.get("api_key")
+        host = host or kwargs.get('host')
+        pool_threads = pool_threads or kwargs.get("pool_threads")
+
+        self._config = Config(api_key=api_key, host=host, **kwargs)
+        
+        api_client = ApiClient(configuration=self._config.OPENAPI_CONFIG, pool_threads=pool_threads)
+        api_client.user_agent = get_user_agent()
+        self._api_client = api_client
+        self._vector_api = VectorOperationsApi(api_client=api_client)
+    
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self._api_client.close()
 
```

Comment:
These methods were the only thing I needed to add back as a consequence changing `Index` to not subclass `ApiClient` to satisfy one test (using `with Index(...) as index` syntax). The [python language docs](https://docs.python.org/3/reference/compound_stmts.html#the-with-statement) show tthese must be implemented in order to use the `with` syntax.